dbms/src/Storages/KVStore/Read/LearnerReadWorker.cpp

```diff
@@ -202,11 +202,19 @@ void LearnerReadWorker::recordReadIndexError(RegionsReadIndexResult & read_index
             auto region_status = RegionException::RegionReadStatus::OTHER;
             if (region_error.has_epoch_not_match())
             {
-                extra_msg = fmt::format(
-                    "read_index_resp error, region_id={} version={} conf_version={}",
-                    region_id,
-                    kvstore->getRegion(region_id)->version(),
-                    kvstore->getRegion(region_id)->confVer());
+                auto kv_region = kvstore->getRegion(region_id);
+                if (kv_region)
+                {
+                    extra_msg = fmt::format(
+                        "read_index_resp error, region_id={} version={} conf_version={}",
+                        region_id,
+                        kv_region->version(),
+                        kv_region->confVer());
+                }
+                else
+                {
+                    extra_msg = fmt::format("read_index_resp error, region_id={} does not exist", region_id);
+                }
```

Comment:
I think we should use the region in `regions_snapshot` to get the `version` and `confVersion` to the logging message?

Reply:
or get it directly from the protobuf struct? https://github.com/pingcap/kvproto/blob/c0689aded40cd69294399a7603544fc5336cc845/proto/errorpb.proto#L81-L87

Reply:
The information in `regions_snapshot` will be printed in an outer exception. However, we can still add it here.

Reply:
We build the read index according to the region meta in `regions_snapshot`. The region meta in `kvstore` could be updated during the learner read.
Which message do we want to show in the error?

Reply:
In doCopLearnerRead, the region info from cop differs from the region info from KVStore. The KVStore has the most up-to-date region epoch, so there is a very large chance that the region info from cop is obsolete, we add log to check that. See https://github.com/pingcap/tiflash/issues/8253